api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.openmrs.Condition;
+import org.openmrs.Patient;
+import org.openmrs.api.APIException;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.ConditionDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * This class implements the {@link ConditionService} interface
+ * It defines the methods to handle the condition domain object
+ *
+ * @since 2.2
+ */
+@Transactional(readOnly = true)
```

Comment:
Our convention has been not to specify the readonly attribute at the class level. But rather do it at the methods that are read only. Do you mind doing the same for the sake of consistency? 😊

Reply:
Okay! I'll do that right away!

Reply:
Did you look into this?

Reply:
@dkayiwa I looked into this. I implemented accordingly. However, based on @wluyima comments, I implemented a reverse of this current implementation you are asking for. 
At any rate, I'll revert right away